### PR TITLE
Feature/add port unique ptr return ref

### PR DESCRIPTION
**Pull request by r-englund**
Made the addPort(std::unique_ptr<Inport|Outport>) return a reference, now it can be used as addPort(T&)

### CODE REVIEW

File: include/inviwo/core/processors/processor.h [hunk 1]
```diff
@@ -397,6 +395,39 @@ class IVW_CORE_API Processor : public PropertyOwner,
...
+}
+
+/**
+ * Add port to processor.
+ * @note Port group is a concept for event propagation. Currently only used for
+ * ResizeEvents, which only propagate from outports to inports in the same port group
+ * @param port to add
+ * @param portGroup name of group to propagate events through (defaults to "default")
+ */
+template <typename T>
```
Comment (ResearchDaniel) [inline include/inviwo/core/processors/processor.h:426]: Documentation should be moved to definition

File: include/inviwo/core/processors/processor.h [hunk 2]
```diff
@@ -336,6 +336,14 @@ class IVW_CORE_API Processor : public PropertyOwner,
     virtual void serialize(Serializer& s) const override;
     virtual void deserialize(Deserializer& d) override;
 
+    // Assume ownership of port, needed for dynamic ports
+    template <typename T, typename std::enable_if_t<std::is_base_of<Inport, T>::value, int> = 0>
```
Comment (ResearchDaniel) [inline include/inviwo/core/processors/processor.h:346]: Could add essentially the same documentation as the other addPort functions

**Review by ResearchDaniel — APPROVED** [review]
Some documentation could be moved/added. Looks good 👍

File: include/inviwo/core/processors/processor.h (hunk 1 above)
Comment (r-englund) [reply include/inviwo/core/processors/processor.h:426]: It was duplicated documentation, forgot to remove it after moving the impl out of the class declaration

**Pull request merged**